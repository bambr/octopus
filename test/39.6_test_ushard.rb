#!/usr/bin/ruby
# coding: utf-8

$: << File.dirname($0)
require '39_test_ushard'

$one.create_shard 1, :POR, "one"
$one.create_object_space 0, :shard => 1, :index => DEFIDX
$one.insert [1,"one"], :shard => 1

# join two into shard 1
$two.create_shard 1, :POR, "one", "two"
sleep 0.3

$one.insert [1, "One"], :shard => 1
$two.insert [1, "Two"], :shard => 1
$two.insert [2, "Two"], :shard => 1
wait_for { $two.select_nolog(2, :shard => 1).length > 0 }

$one.select 1, :shard => 1
$two.select 1, :shard => 1

$two_env.env_eval do
  stop
  puts `./octopus --cat 00000000000000000001.snap`.gsub(/ tm:\d+(\.\d+)? /, ' ')
  puts `./octopus --cat 00000000000000000002.xlog`.gsub(/ tm:\d+(\.\d+)? /, ' ')
end
