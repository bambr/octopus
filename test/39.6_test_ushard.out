# one.ping()
:pong

# two.ping()
:pong

# one.meta(shard 1 create por)
ok

# one.meta(shard 1 obj_space 0 create tree unique string 0)
ok

# one.insert([1, "one"], {:shard=>1})
1

# two.meta(shard 1 add_replica two)
ok

# one.insert([1, "One"], {:shard=>1})
2

# two.insert([1, "Two"], {:shard=>1})
2

# two.insert([2, "Two"], {:shard=>1})
1

# one.select(1, {:shard=>1})
[["\x01\x00\x00\x00", "Two"]]

# two.select(1, {:shard=>1})
[["\x01\x00\x00\x00", "Two"]]

lsn:2 scn:-1 t:sys/snap_initial ver:0 count:1 flags:0x00000000
lsn:2 shard:1 scn:5 t:sys/shard_create SHARD_CREATE shard_id:1 POR Box count:1 run_crc:0xcaef1b66 master:one repl:two
lsn:2 shard:1 scn:5 t:snap/usr240 CREATE_OBJECT_SPACE n:0 flags:00000003 cardinalty:0 PK: i:0 min_tuple_cardinality:0 cardinality:1 type:5 unique:1 field0:{index:0 type:7 sort:1}
lsn:2 shard:1 scn:5 t:snap/snap_data n:0 <1:"\x01\x00\x00\x00", "One">
lsn:2 shard:1 scn:5 t:sys/shard_final 
lsn:2 scn:-1 t:sys/snap_final 
lsn:2 shard:1 scn:5 t:wal/usr13 INSERT n:0 flags:00000000 <1:"\x01\x00\x00\x00", "One">
lsn:3 shard:1 scn:6 t:wal/usr13 INSERT n:0 flags:00000000 <1:"\x01\x00\x00\x00", "Two">
lsn:4 shard:1 scn:7 t:wal/usr13 INSERT n:0 flags:00000000 <2:"\x02\x00\x00\x00", "Two">
